src/it/hentaiarchive/src/eu/kanade/tachiyomi/extension/it/hentaiarchive/HentaiArchive.kt

```diff
@@ -0,0 +1,174 @@
+        .build()
+
+    override val client = super.client.newBuilder()
+        .addInterceptor { chain ->
+            val request = chain.request()
+            val url = request.url.toString()
+            if (url.contains("picsarchive1.b-cdn.net")) {
+                return@addInterceptor chain.proceed(request.newBuilder().headers(cdnHeaders).build())
+            }
+            chain.proceed(request)
+        }
+        .build()
+
+    override val supportsLatest = false
+
+    // ============================== Popular ===============================
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/category/hentai-recenti/page/$page", headers)
+
+    override fun popularMangaSelector() = "div.posts-container article"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        // Extract and set the URL without domain
+        setUrlWithoutDomain(element.selectFirst("a.entire-meta-link")!!.absUrl("href"))
+
+        // Extract and set the title
+        title = element.selectFirst("span.screen-reader-text")!!.text()
+
+        // Extract and set the thumbnail URL
+        thumbnail_url = element.selectFirst("span.post-featured-img img.wp-post-image")?.absUrl("data-nectar-img-src")
+    }
+
+    override fun popularMangaNextPageSelector() = "nav#pagination a.next.page-numbers"
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int): Request {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // Define the array of specific queries
+        // val specialQueries = arrayOf("hentai-senza-censura", "hentai-tette-grosse", "hentai-milf", "hentai-studentesse", "hentai-full-color", "hentai-ita-controllo-mentale", "hentai-ita-bikini", "hentai-culi-grossi", "hentai-pelle-scura", "hentai-insegnanti", "hentai-tradimento", "hentai-mostri", "hentai-sesso-anale", "doujinshi", "hentai-sex-toys", "hentai-superdotati", "hentai-monster-girl", "fumetti-porno", "hentai-ita-harem", "hentai-futanari", "hentai-uomini-maturi", "hentai-ita-domestiche", "hentai-esibizioniste", "hentai-famiglia", "hentai-fantasy", "hentai-femdom", "hentai-follia", "hentai-formose", "hentai-pelose", "hentai-lesbiche", "hentai-yaoi", "hentai-x-ray", "hentai-gola-profonda", "hentai-ita-sesso-estremo", "hentai-gyaru", "hentai-ita-infermiere")
+
+        // Default behavior
+        return GET("$baseUrl/page/$page/?s=$query")
```

Comment:
use `HttpUrl.Builder` to properly encode the query.

Reply:
cant do it

Reply:
```
val url = "$baseUrl/page/$page/".toHttpUrl().newBuilder()
    .addQueryParameter("s", query)
    .build()
```